src/Hub.cpp

```diff
@@ -252,6 +253,28 @@ void Hub::AssignVespeneHarvester(const sc2::Unit& refinery_) {
     worker->GatherVespene(refinery_);
 }
 
+void Hub::DistrubuteMineralWorker(const sc2::Unit* unit_) {
+    auto patches = gAPI->observer().GetUnits(sc2::IsVisibleMineralPatch(), sc2::Unit::Alliance::Neutral);
```

Comment:
A small issue: we could have e.g. macro hatcheries which shouldn't be saturated.

Reply:
Yes, when I work on Issue #42 ("Support build of macro hatchery") in order to build Macro Orbitals, I'll test and make sure this works with macros. (eventually may also need to support long-distance mining)

Reply:
Have changed this to iterate through Expansions, so won't have macro issue anymore.